### PR TITLE
FEATURE: Upload component

### PR DESCRIPTION
**Pull request by kitsunet**
Adds a component to facilitate asset uploads to Neos.
The component is used inside the Image editor but can be re-used
for other editors.

Fixes: #874

### CODE REVIEW

File: packages/neos-ui-editors/src/Library/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import AssetUpload from './AssetUpload';
+
+export default () => ({
+    AssetUpload
+});
```
Comment (skurfuerst) [inline packages/neos-ui-editors/src/Library/index.js]: This is wrong; you need to do an "export AssetUpload" here :)

A named import in the form `import {....}` is *never* accessing the default import; they are two totally different things.

File: packages/neos-ui-editors/src/Editors/Image/index.js [hunk 2]
```diff
@@ -220,21 +220,16 @@ export default class ImageEditor extends Component {
     }
 
     upload(files) {
```
Comment (dimaip) [inline packages/neos-ui-editors/src/Editors/Image/index.js]: files defined but never used

File: packages/neos-ui-editors/src/Editors/Image/index.js [hunk 3]
```diff
@@ -220,21 +220,16 @@ export default class ImageEditor extends Component {
...
-        const {commit, siteNodePath} = this.props;
+        const {commit} = this.props;
         const {isImageCropperOpen} = this.state;
 
-        const siteNodeName = siteNodePath.match(/\/sites\/([^/@]*)/)[1];
-
-        return uploadAsset(files[0], siteNodeName).then(res => {
-            this.setState({image: res}, () => {
-                commit(res.object);
+        this.setState({image: res}, () => {
```
Comment (dimaip) [inline packages/neos-ui-editors/src/Editors/Image/index.js]: res is undefined

**Comment by kitsunet** [comment]
@dimaip this has been updated, retested, linted and should just work now 😆  Please review

**Pull request merged**